R/check_spatvector.R

```diff
@@ -0,0 +1,145 @@
+#' Check SpatVector Properties
+#'
+#' This function checks if a given object is a valid SpatVector polygon with
+#' appropriate coordinate system and size constraints, and validates its overlap
+#' with known country boundaries.
+#'
+#' @param x Object to check.
+#' @param check_size Logical. Whether to check if the area is within reasonable bounds.
+#' Default is TRUE.
+#'
+#' @return The input SpatVector, possibly reprojected to EPSG:4326.
+#'
+#' @details
+#' The function checks:
+#' - If input is a SpatVector polygon
+#' - If coordinate system is properly defined
+#' - If polygons are valid and non-empty
+#' - If the area overlaps with known country boundaries
+#' - Optionally checks if area is within reasonable bounds
+#' (larger than 10,000 ha but smaller than approximately Brazil)
+#'
+#' @import terra
+#'
+#' @references
+#' Zillah Calle (2023)
+#' Jonas van Duijvenbode (2023)
+#'
+#' @examples
+#' \dontrun{
+#' shape <- vect("path/to/shapefile.shp")
+#' validated_shape <- check_spatvector(shape)
+#' }
+#'
+#' @export
+check_spatvector <- function(x, check_size = TRUE) {
```

Comment:
As we're checking an object that's probably a SpatVector calling the input variable x is not the most descriptive. Maybe something like "polygon" would be better.

Reply:
this one i changed!